File: C/easy/remove_element.c
/**************************************************************Method 1**************************************************************/
void swap(int *a, int *b){
    
    int temp;
    
    temp = *a;
    *a = *b;
    *b = temp;
}

int removeElement(int* nums, int numsSize, int val){
    
    int i = 0;
    int end = numsSize;
    
    while(i < end){
        
        if(nums[i] == val){
        
            swap(&nums[i], &nums[end - 1]);
            end--;    
        }
            
     else{i++;}        
        
    }
    
    return end;

}
/**************************************************************Method 1**************************************************************/

/**************************************************************Method 2**************************************************************/
int removeElement(int *nums, int numsSize, int val)
{
    int i, j;
    i = 0, j = 0;
    while (i < numsSize)
    {
        int k = 0;
        if (nums[i] == val)
        {
            while (1)
            {
                if (j < numsSize)
                {
                    if (nums[j] == val)
                    {
                        j++;
                    }
                    else
                    {
                        break;
                    }
                }
                else
                {
                    k = 1;
                    break;
                }
            }
            if (k)
            {
                break;
            }
            int temp = nums[i];
            nums[i] = nums[j];
            nums[j] = temp;
        }
        i++;
        j++;
    }
    return i;
}
/**************************************************************Method 2**************************************************************/

File: Python/easy/remove_element.py
# Method 1
class Solution:
    def removeElement(self, nums, val):
        """
        : List[int]
        : int
         -> int
        """
        
        for i in range(nums.count(val)):
            nums.remove(val)
        return len(nums)

# Method 2
class Solution:
    def removeElement(self, nums: List[int], val: int) -> int:
        if len(nums) == 0: return 0
        l = 0
        for i in range(len(nums)):
            if nums[i] != val:
                nums[l] = nums[i]
                l += 1
        return l
